btree/btree.c

```diff
@@ -4499,9 +4499,7 @@ int  btree_sibling_first_key(struct m0_btree_oimpl *oi, struct td *tree,
 			return 0;
 		}
 	}
-	s->s_rec.r_flags = M0_BSC_KEY_NOT_FOUND;
-	return 0;
-
+	return M0_ERR(-ENOENT);
```

Comment:
you can return M0_BSC_KEY_NOT_FOUND instead of M0_ERR(-ENOENT) as we need to return linux codes only value for M0_BSC_KEY_NOT_FOUND is assigned equal to ENOENT. so that returning M0_BSC_KEY_NOT_FOUND will provide more readability and we will return linux codes only.

Please do the same changes for rest of places.

Reply:
in btree.h value for M0_BSC_KEY_NOT_FOUND  is assigned equal to ENOENT, therefore I suggested to return M0_BSC_KEY_NOT_FOUND. But if M0_BSC_* are purposed for r_flags and we should return ENOENT , above implementation is correct. I will make similar changes in put and delete as in put I am returning M0_BSC_KEY_NOT_FOUND instead of ENOENT.